### ISSUE TITLE
Synchronous deployment - invalid return code

### ISSUE THREAD

**Issue by kucharzyk-sebastian** [aliyun/Resource-Orchestration-Service-Cloud-Development-Kit#35]
We've recently noticed that the command `ros-cdk deploy --sync` sometimes returns `0` error code even when it failed. I'd expect it to return non-zero code in such cases. Here's a minimalistic example that let us discover the problem:

`lib/infrastructure-stack.ts`:
```typescript
import * as ros from "@alicloud/ros-cdk-core";
import * as rds from "@alicloud/ros-cdk-rds";


export class InfrastructureStack extends ros.Stack {
    constructor(scope: ros.Construct, id: string) {
      super(scope, id);
      new rds.DBInstance(this, id, {
        engine: "MySQL",
        dbInstanceClass: "mysql.n1.micro.1",
        dbInstanceStorage: 50,
        engineVersion: "8.0",
        securityIpList: "0.0.0.0/0",
        dbInstanceStorageType: "cloud_essd",
        vpcId: "<MASKED>",
        vSwitchId: "<MASKED>",
        payType: "PayAsYouGo",
        dbInstanceDescription: "IPC MySQL DB",
        allocatePublicConnection: true,
        multiAz: true,
      });
    }
  }
```

`bin/infrastructure.ts`:
```typescript
#!/usr/bin/env node
import * as ros from "@alicloud/ros-cdk-core";
import { InfrastructureStack } from "../lib/infrastructure-stack";

const app = new ros.App({ outdir: "./cdk.out" });
new InfrastructureStack(app, "invalid-return-code-poc");
```

logs:
```shell
seku@PC:~/dev/invalid-return-code-poc$ ros-cdk deploy --sync
invalid-return-code-poc: deploying...
|invalid-return-code-poc |2022-02-28T12:06:13 | DELETE_IN_PROGRESS   | ALIYUN::RDS::DBInstance |  | invalid-return-code-poc
 ✅ The deployment(sync deploy stack) has finished!
status: CREATE_FAILED
StatusReason: Resource CREATE failed: ResponseException: resources.invalid-return-code-poc: The Specified Zone not supported. Code: InvalidZone.NotSupported RequestId: <MASKED>
StackId: <MASKED>

seku@PC:~/dev/invalid-return-code-poc$ echo $?
0
```

**Comment by baichenxu3** [comment]
We will fix this problem and return code 2 for operations that fail to create after optimization

**Comment by kucharzyk-sebastian** [comment]
I can confirm the problem is solved with ros-cdk-cli 1.0.19. Thanks!
